integrations/oci/sdk/runtime/src/main/java/io/helidon/integrations/oci/sdk/runtime/OciExtension.java

```diff
@@ -109,44 +114,124 @@
+     * If the implementation is unable to find this file, then a fallback mechanism will be used to find it in the configuration
+     * found in the {@link InjectionServices#globalBootstrap()}, using a top-level attribute key named
+     * {@value OciConfigBlueprint#CONFIG_KEY}.
+     * <p>
+     * The final fallback mechanism will use an {@code auto} authentication strategy - see {@link OciConfigBlueprint} for details.
      *
      * @return the bootstrap oci config bean
+     * @see OciConfigBlueprint
+     * @see #ociConfigSupplier
      */
     public static OciConfig ociConfig() {
-        Optional<Bootstrap> bootstrap = InjectionServices.globalBootstrap();
-        if (bootstrap.isEmpty()) {
-            LOGGER.log(System.Logger.Level.DEBUG, "No bootstrap - using default oci config");
-            return DEFAULT_OCI_CONFIG_BEAN.get();
-        }
+        io.helidon.common.config.Config config = configSupplier().get();
+        if (!config.exists()
+                || !(config.get(OciAuthenticationDetailsProvider.KEY_AUTH_STRATEGY).exists()
+                  || config.get(OciAuthenticationDetailsProvider.KEY_AUTH_STRATEGIES).exists())) {
+            // fallback
+            Optional<Bootstrap> bootstrap = InjectionServices.globalBootstrap();
+            if (bootstrap.isEmpty()) {
+                LOGGER.log(System.Logger.Level.DEBUG, "No bootstrap - using default oci config");
+                return DEFAULT_OCI_CONFIG_BEAN.get();
+            }
 
-        Config config = bootstrap.get().config().orElse(null);
-        if (config == null) {
-            LOGGER.log(System.Logger.Level.DEBUG, "No config in bootstrap - using default oci config");
-            return DEFAULT_OCI_CONFIG_BEAN.get();
-        }
+            config = bootstrap.get().config().orElse(null);
+            if (config == null) {
+                LOGGER.log(System.Logger.Level.DEBUG, "No config in bootstrap - using default oci config");
+                return DEFAULT_OCI_CONFIG_BEAN.get();
+            }
 
-        config = config.get(OciConfig.CONFIG_KEY);
-        if (!config.exists()) {
-            LOGGER.log(System.Logger.Level.DEBUG, "No oci config in bootstrap - using default oci config");
-            return DEFAULT_OCI_CONFIG_BEAN.get();
+            config = config.get(OciConfig.CONFIG_KEY);
+            if (!config.exists()) {
+                LOGGER.log(System.Logger.Level.DEBUG, "No oci config in bootstrap - using default oci config");
+                return DEFAULT_OCI_CONFIG_BEAN.get();
+            }
         }
 
-        LOGGER.log(System.Logger.Level.DEBUG, "Using specified oci config");
         return OciConfig.create(config);
     }
 
+    /**
+     * The supplier for the raw config-backed by the OCI config source(s).
+     *
+     * @return the supplier for the raw config-backed by the OCI config source(s)
+     * @see #ociAuthenticationProvider()
+     */
+    public static Supplier<io.helidon.common.config.Config> configSupplier() {
```

Comment:
This is a static method, yet it totally ignores thread safety. Please fix.

Reply:
Also this is why `LazyValuel` exists in Helidon - you can directly use that type and this whole method is not needed

Reply:
`LazyValue` does not support reset, so I need to do this when the config file name is changed (to support testing).

Reply:
pushed